redisinsight/ui/src/services/wsService.ts

```diff
@@ -0,0 +1,53 @@
+import { io } from 'socket.io-client'
+import { getProxyPath } from 'uiSrc/utils'
+import { CustomHeaders } from 'uiSrc/constants/api'
+import { getConfig } from 'uiSrc/config'
+
+const riConfig = getConfig()
+
+export type WsParams = {
+  forceNew?: boolean;
+  token?: string;
+  reconnection?: boolean;
+  query?: Record<string, any>;
+  extraHeaders?: Record<string, any>;
+  path?: string;
+}
+
+export function wsService(
+  wsUrl: string,
+  { forceNew = true, token, reconnection, query, extraHeaders, path = getProxyPath() }: WsParams,
+  passTokenViaHeaders: boolean = true,
```

Comment:
I see that we pass `envDependent` here, but what does it do? Looking at the implementation looks like we switch from query params to headers for that token, but was wondering what is the general idea here.

Reply:
the idea/issue is, that WS connection works different in browser than on the desktop, which is why we have env dependent implementation

Reply:
So, `envDependent` is used for other browser/desktop issues?

Reply:
Yes, that is what we use for "feature" flag ... I know it is not intuitive namming, but there are reasons why

Reply:
Thanks!